### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by GraceHtet**
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user
<hr/>
<table>
<thead>
<th>Before</th>
<th>After</th>
</thead>
<tbody>
<tr>
<td>

</td>
<td>

</td>
<tr>
</tbody>
</table>

Integration specs
Use Capybara to write integration tests for each view in project.

- User index page:

> I can see the username of all other users.
> I can see the profile picture for each user.
> I can see the number of posts each user has written.
> When I click on a user, I am redirected to that user's show page.

- user show page:

> I can see the user's profile picture.
> I can see the user's username.
> I can see the number of posts the user has written.
> I can see the user's bio.
> I can see the user's first 3 posts.
> I can see a button that lets me view all of a user's posts.
> When I click a user's post, it redirects me to that post's show page.
> When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:

> I can see the user's profile picture.
> I can see the user's username.
> I can see the number of posts the user has written.
> I can see a post's title.
> I can see some of the post's body.
> I can see the first comments on a post.
> I can see how many comments a post has.
> I can see how many likes a post has.
> I can see a section for pagination if there are more posts than fit on the view.
> When I click on a post, it redirects me to that post's show page.

- Post show page:

> I can see the post's title.
> I can see who wrote the post.
> I can see how many comments it has.
> I can see how many likes it has.
> I can see the post body.
> I can see the username of each commentor.
> I can see the comment each commentor left.

### CODE REVIEW

File: spec/system/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      assert_text @tom.posts.first.text
+    end
+
+    scenario 'I can see the username of each commentor and the comment each commentor left.' do
+      visit user_post_path(@tom, @tom.posts.first)
+
+      assert_text "#{@tom.posts.first.comments.first.author.name}:#{@tom.posts.first.comments.first.text}"
+    end
+  end
+end
```
Comment (Gambit142) [inline spec/system/post_spec.rb]: - [ ] Based on the requirements, each page should have a test file that contains all tests written for that page. This means that you should have separate test files for the `Post#Index` page and the `Post#Show` page and write the tests in them based on the requirement for that page. 👍

File: spec/system/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    scenario 'When I click on a post, it redirects me to that post show page' do
+      visit user_posts_path(@tom.id)
+      within("#post-card-#{@post3.id}") do
+        click_on @post3.title
+      end
+      expect(page).to have_current_path(user_post_path(@tom.id, @post3.id))
+    end
+  end
+
```
Comment (Gambit142) [inline spec/system/post_spec.rb]: - [ ] For the `Post#Index` page, you forgot to write a test that asserts a _**User**_ can see how many likes a _**Post**_ has. Kindly include this when you have separated the tests into several files. 👍 😎

File: spec/system/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    end
+
+    scenario 'When I click a user post, it redirects me to that post show page' do
+      visit user_path(@tom)
+      click_link @post1.title.to_s
+
+      assert_current_path user_post_path(@tom, @post1)
+    end
+  end
+end
```
Comment (Gambit142) [inline spec/system/user_spec.rb]: - [ ] Based on the requirements, each page should have a test file that contains all tests written for that page. This means that you should have separate test files for the `User#Index` page and the `User#Show` page and write the tests in them based on the requirement for that page. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @GraceHtet @Fahim-Khan-P 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Use of `capybara` gem ✔️ 
- Wrote integration tests for the index and show pages for the `User` model ✔️
- Wrote integration tests for the index and show pages for the `Post` model ✔️  

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @GraceHtet, and @Fahim-Khan-P👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- All tests are passed. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**